server/events/project_locker.go

```diff
@@ -87,7 +91,9 @@ func (p *DefaultProjectLocker) TryLock(log logging.SimpleLogging, pull models.Pu
 	return &TryLockResponse{
 		LockAcquired: true,
 		UnlockFn: func() error {
-			_, err := p.Locker.Unlock(lockAttempt.LockKey)
+			// TODO monikma #8 this will be called if there was a plan error and the lock was automatically dropped;
+			// Should we assure dequeuing of the next PR here too?
+			_, _, err := p.Locker.Unlock(lockAttempt.LockKey, false)
```

Comment:
Original discussion: https://github.com/runatlantis/atlantis/pull/1923#discussion_r955461604

@monikma 
> Yes, what do you think, what would make more sense?
> 
> 1. Just drop everything and wait for the lock owner to fix their plan and acquire the lock again, or
> 2. Say "your loss, now you are out of the queue", and dequeue the next PR, or
> 3. Quickly re-acquire the lock for this PR so that the owner can plan as soon as they fixed it
> 
> .. we have not really decided. I guess 1 is not a good option as then a third person could accidentally get the lock.. also 3 would not be fair since fixing can take time. So I guess 2. Yes, we have not implemented it yet, we were waiting for some general feedback on the functionality first :)

Reply:
Option 1: this would mean anyone can obtain the lock and get in front of the queue in the meantime
Option 3: That was our initial option but we were not sure why we release the lock in case of plan error in the first place.
Maybe we can also have Option 4: feature flag to either go with Option 2 (dequeue on plan error) or Option 3 (always keep the lock on plan error).

@nitrocode any thoughts?

Reply:
cc: @runatlantis/maintainers and @krrrr38 for visibility

Reply:
I like Option 4 that way the behavior is configurable to the user workflow that they prefer.

Reply:
@jamengual since this PR is getting very large, do you think it makes sense to introduce such feature flag in a follow-up, and for now just give the lock to the next in queue?

Reply:
I think we can add them in another PR but the feature if approved will not be released if the flags are not added, so we will have to have both PRs

Reply:
I see, then I will consider it in this PR if you don't mind the size.